pkg/agent/agent_test.go

```diff
@@ -708,3 +712,261 @@ func TestRestorePortConfigs(t *testing.T) {
 		})
 	}
 }
+
+func TestSetOVSDatapath(t *testing.T) {
+	tests := []struct {
+		name          string
+		expectedCalls func(m *ovsconfigtest.MockOVSBridgeClient)
+		expectedErr   string
+	}{
+		{
+			name: "fail to read OVS bridge other_config",
+			expectedCalls: func(m *ovsconfigtest.MockOVSBridgeClient) {
+				m.EXPECT().GetOVSOtherConfig().Return(nil, ovsconfig.InvalidArgumentsError("failed to read OVS bridge other_config"))
+			},
+			expectedErr: "failed to read OVS bridge other_config",
+		},
+		{
+			name: "fail to set OVS bridge datapath_id",
+			expectedCalls: func(m *ovsconfigtest.MockOVSBridgeClient) {
+				m.EXPECT().GetOVSOtherConfig().Return(map[string]string{}, nil)
+				m.EXPECT().SetDatapathID(mock.Any()).Return(ovsconfig.InvalidArgumentsError("failed to set OVS bridge datapath_id"))
+			},
+			expectedErr: "failed to set OVS bridge datapath_id",
+		},
+		{
+			name: "datapath-id exists in other_config on OVS bridge",
+			expectedCalls: func(m *ovsconfigtest.MockOVSBridgeClient) {
+				m.EXPECT().GetOVSOtherConfig().Return(map[string]string{
+					ovsconfig.OVSOtherConfigDatapathIDKey: "datapathId",
+				}, nil)
+			},
+		},
+		{
+			name: "generate and set datapath ID for OVS bridge",
+			expectedCalls: func(m *ovsconfigtest.MockOVSBridgeClient) {
+				m.EXPECT().GetOVSOtherConfig().Return(map[string]string{}, nil)
+				m.EXPECT().SetDatapathID(mock.Any())
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			controller := mock.NewController(t)
+			mockOVSBridgeClient := ovsconfigtest.NewMockOVSBridgeClient(controller)
+			initializer := newAgentInitializer(mockOVSBridgeClient, nil)
+			tt.expectedCalls(mockOVSBridgeClient)
+
+			err := initializer.setOVSDatapath()
+			if tt.expectedErr != "" {
+				assert.ErrorContains(t, err, tt.expectedErr)
+			} else {
+				require.NoError(t, err)
+			}
+		})
+	}
+}
+
+func mockIPsecPSKEnv(name string) func() {
+	os.Setenv(ipsecPSKEnvKey, name)
+	return func() { os.Unsetenv(ipsecPSKEnvKey) }
+}
+
+func TestReadIPSecPSK(t *testing.T) {
+	tests := []struct {
+		name        string
+		expectedErr string
+	}{
+		{
+			name:        "IPsec PSK env variable not set",
+			expectedErr: "IPsec PSK environment variable 'ANTREA_IPSEC_PSK' is not set or is empty",
+		},
+		{
+			name: "IPsec PSK env variable set",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			initializer := &Initializer{
+				networkConfig: &config.NetworkConfig{
+					IPsecConfig: config.IPsecConfig{},
+				},
+			}
+			if tt.expectedErr == "" {
+				defer mockIPsecPSKEnv("key")()
+			}
```

Comment:
Define input parameter to control how the test runs, instead of using expected output to control it. Otherwise the test table itself is confusing: why the first would fail and the second would succeed when there is no difference.